src/domain/token/InitTokenOptions.ts

```diff
@@ -7,5 +7,5 @@ export interface InitTokenOptions {
   tokenAddress: string;
   moonshot: Moonshot;
   curveType?: CurveType;
-  provider: ethers.JsonRpcProvider;
+  provider?: ethers.JsonRpcProvider;
```

Comment:
this is actually not needed anymore. It is replaced by signer in Token class. Did not remove it, to not break the interface, let me know if I should remove it.

Reply:
good to know, you can add @depricated tag, we will remove it in the next major release